crates/net/eth-wire-types/src/receipts.rs

```diff
@@ -17,15 +18,33 @@ pub struct GetReceipts(
 
 /// The response to [`GetReceipts`], containing receipt lists that correspond to each block
 /// requested.
-#[derive(Clone, Debug, PartialEq, Eq, RlpEncodableWrapper, RlpDecodableWrapper, Default)]
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
 #[add_arbitrary_tests(rlp)]
-pub struct Receipts(
+pub struct Receipts<T = Receipt>(
     /// Each receipt hash should correspond to a block hash in the request.
-    pub Vec<Vec<ReceiptWithBloom<Receipt>>>,
+    pub Vec<Vec<ReceiptWithBloom<T>>>,
 );
 
+impl<T: RlpEncodableReceipt> alloy_rlp::Encodable for Receipts<T> {
+    #[inline]
+    fn encode(&self, out: &mut dyn alloy_rlp::BufMut) {
+        self.0.encode(out)
+    }
+    #[inline]
+    fn length(&self) -> usize {
+        self.0.length()
+    }
+}
+
+impl<T: RlpDecodableReceipt> alloy_rlp::Decodable for Receipts<T> {
+    #[inline]
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        alloy_rlp::Decodable::decode(buf).map(Self)
```

Comment:
what's the `map(Self)` for?

Reply:
this decodes the inner `Vec` type and the `map(Self)` wraps that into `Receipts()`